### DISCUSSION TITLE
mapStores returns undefined on vue2

### DISCUSSION THREAD

**Discussion by Archer-Thane** [vuejs/pinia#999] [Help and Questions]
Hey there. I want to use pinia along side vue 2 and options api. 
So I followed the instructions and do this:
![Screen Shot 1400-11-06 at 13 05 23](https://user-images.githubusercontent.com/17456531/151138729-50d5fbb2-f349-4ab0-ba1b-a4f1b94ff02f.png)
and then make a file for the store putting logic there:
![Screen Shot 1400-11-06 at 13 05 46](https://user-images.githubusercontent.com/17456531/151138798-bf6d6a95-26a2-4250-9672-ab5ee26ecb9e.png)
finally call it inside my component using ...mapStores
![Screen Shot 1400-11-06 at 13 06 10](https://user-images.githubusercontent.com/17456531/151138862-190017d6-cfc4-4907-ab83-8541b048b3a2.png)
but then it returns me undefined. can any one help? thanks alot

**Comment by mkidsc1603** [comment]
`this.useRegisterStore`  => `this.registerStore` or `this.register` ?

I suggested that file's name same as store name.

**Reply by posva** [reply]
yes, it's `this.registerStore` by default: https://pinia.vuejs.org/api/modules/pinia.html#mapstores
